File: SWIG_tutorial/example2/ex2.c
/****************************************************************
* Definition of the functions declared in the header file
* @author: Surya Paruchuri
* @date: December 23 2017.
* (c) Surya Paruchuri
*****************************************************************/
#include<stdio.h>
#include"ex2.h"

void printSum(int a, int b){ printf("The sum of %d and %d is : %d\n",a,b,a+b); }

void storeSum(int a, int b, int* result){ *result = a+b; return; }

File: SWIG_tutorial/example2/ex2.py
###############################################################
# Test Script to demonstrate the use of testmaps.i in SWIG to
# map inputs and outputs.
# @author: Surya Paruchuri
# @date: December 23 2017.
###############################################################

from swig_ex2 import *

def main():
	print("++++++++++ Main function execution begins ++++++++++++")
	print("++++ calling printSum ++++")
	printSum(5,10)
	print("++++ calling storeSum +++++")
	storeVar = storeSum(115,230)
	print("The value returned is: %s"%storeVar)
	

if __name__== "__main__": main()
